### PR TITLE
chore(user-profile): Remove migration worker and make user profile strict

### PR DESCRIPTION
**Pull request by saevarma**
Corresponding Helm cleanup PR
https://github.com/island-is/helm/pull/785

## What

* Removing all migration worker related code
* Make user profile service TS strict

## Why

* Migration was a one time job which is done.
* TS strict helps detect issues during build time.

## Screenshots / Gifs

N/A

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Updated email and SMS verification logic to allow optional fields and improve flexibility.
  - Enhanced user profile service to handle default values more effectively.

- **Bug Fixes**
  - Fixed inconsistencies in user profile data handling, ensuring reliable data updates and error handling.

- **Refactor**
  - Simplified configuration and removed deprecated worker-related settings.
  - Streamlined code by consolidating configuration settings and removing unnecessary variables.

- **Chores**
  - Removed outdated migration tables and worker configurations from the database and project settings.
  - Updated environment configurations to remove unused variables and settings.

- **Documentation**
  - Added comments and documentation for new features and refactored code to improve clarity.

### CODE REVIEW

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Comment by andes-it** [comment]
This PR closed its doors. Sat around for a day after the stale tag, then reckoned it's not worth the fuss.

File: apps/services/user-profile/src/config.ts [hunk 1]
```diff
@@ -19,10 +18,6 @@ export const UserProfileConfig = defineConfig({
       migrationDate: new Date(
         env.optional('USER_PROFILE_MIGRATION_DATE') ?? '2024-04-12',
       ),
```
Comment (baering) [inline apps/services/user-profile/src/config.ts:20]: Should this be removed as well?

**Review by baering — APPROVED** [review]

File: apps/services/user-profile/src/config.ts (hunk 1 above)
Comment (saevarma) [reply apps/services/user-profile/src/config.ts:20]: No, we use this check if we are able to share the data. We check if last nudge is newer than this date to know if the user has seen the new screens due to consent statement on them.

**Pull request merged**